universal-application-tool-0.0.1/app/auth/FakeAdminClient.java

```diff
@@ -0,0 +1,38 @@
+package auth;
+
+import controllers.admin.routes;
+import java.util.Optional;
+import javax.inject.Inject;
+import org.pac4j.core.client.IndirectClient;
+import org.pac4j.core.credentials.AnonymousCredentials;
+import org.pac4j.core.util.HttpActionHelper;
+
+public class FakeAdminClient extends IndirectClient {
+
+  private ProfileFactory profileFactory;
+
+  @Inject
+  public FakeAdminClient(ProfileFactory profileFactory) {
+    this.profileFactory = profileFactory;
+  }
+
+  @Override
+  protected void internalInit() {
+    defaultCredentialsExtractor(
+        (ctx, store) -> {
+          // Double check that we haven't been fooled into allowing this somehow.
+          if (!ctx.getServerName().equals("localhost")) {
+            throw new UnsupportedOperationException(
+                "You cannot create an admin unless you are running locally.");
```

Comment:
nit: "You cannot create a *fake* admin ... " ?

Reply:
Done!